lib/broker/rabbit/queue.ts

```
@@ -14,50 +16,84 @@ export class Queue {
+			'topic',
+			{ durable: true },
+		);
+		const assertedQueue = await this.channel.assertQueue(name, {
+			exclusive: false,
+			durable: true,
+		});
+
+		await this.channel.bindQueue(
+			assertedQueue.queue,
+			assertedExchange.exchange,
+			'',
+		);
+
+		return new Promise<Message>(async (resolve) => {
+			await this.channel.consume(name, (message: ConsumeMessage) => {
+				const eventPayload: Record<string, any> = JSON.parse(
+					String(message.content),
+				);
+				const deliveryInfo: DeliveryInfo = {
+					deliveryTag: String(message.fields.deliveryTag),
+					routingKey: message.fields.routingKey,
+				};
+
+				this.callback(deliveryInfo, eventPayload, message as Message, callback);
+				resolve(message);
 			});
 		});
 	}
 
+	/**
+	 * Callback to create new rabbit context to handle received message
+	 * @param {string} deliveryInfo - info about recived queue message
+	 * @param {Record<string, any>}  payload - actually the body message
+	 * @param {Function} method - next callback to execute anything, like call context (ack, nack...whatever)
+	 * @returns {void} void
+	 *
+	 */
 	private callback(
 		deliveryInfo: DeliveryInfo,
-		properties: any,
 		payload: Record<string, any>,
-		func: Function,
-	) {
+		message: Message,
+		method: (event: Event, context: Context) => void,
+	): void {
 		const eventName = deliveryInfo.routingKey;
 		const event = new Event(eventName, payload);
-	}
-
-	private queueOptions() {
-		return { durable: true };
+		const context = new RabbitContext(this.channel, message);
+		method(event, context);
 	}
 
 	/**
 	 * Returns the full queue name
 	 * @param {string}  routingKey  - queue path string
-	 * @returns {string}
+	 * @returns {string} string
 	 */
-	private queueName(routingKey: string) {
-		return `${this.config.APP_NAME.toLocaleLowerCase()}-#${routingKey.toLocaleLowerCase()}`;
+	private queueName(routingKey: string): string {
+		const splitEventName = routingKey.split('.');
+
+		if (splitEventName.length < 3)
```

Greater than 4 should also raise an error.